blocks/handler.go

```diff
@@ -72,11 +73,11 @@ func (h *Handler) HandleSyncedBlock(ctx context.Context, expHash types.Hash32, p
 	}
 
 	if b.LayerIndex <= types.GetEffectiveGenesis() {
-		return fmt.Errorf("block before effective genesis: layer %v", b.LayerIndex)
+		return fmt.Errorf("%w: block before effective genesis: layer %v", pubsub.ErrValidationReject, b.LayerIndex)
 	}
 
 	if err := ValidateRewards(b.Rewards); err != nil {
-		return fmt.Errorf("%w: %w", errInvalidRewards, err)
+		return fmt.Errorf("%w: %s", pubsub.ErrValidationReject, err.Error())
```

Comment:
There is a new API for this introduced in Go 1.20:

```suggestion
		return errors.Join(pubsub.ErrValidationReject, err)
```

Reply:
i don't think that all errors should be chained by default. error is a part of api, so that it can be handled properly.  nobody cares about errInvalidRewards (should have been just string), thats just an information. pubsub.ErrValidationReject part of api